onnxruntime/core/framework/allocation_planner.cc

```diff
@@ -2267,11 +2248,14 @@ Status DeviceBasedPartitioner::PartitionGraph(const onnxruntime::GraphViewer& gr
         device_types_.push_back(device_type);
         it = device_to_stream.find(device_type);
       }
-      // put the node into the belonging stream
-      if (node_name.empty()) {
-        node_names_by_stream_[it->second].push_back(op_type + std::to_string(op_type_counter[op_type]++));
+      std::string node_name_or_type = node_name;
+      if (node_name_or_type.empty()) {
+        node_name_or_type = op_type + std::to_string(op_type_counter[op_type]++);
+      }
+      if (op_type == "MemcpyToHost" || op_type == "MemcpyFromHost") {
```

Comment:
>op_type == "MemcpyToHost" || op_type == "MemcpyFromHost [](http://example.com/codeflow?start=10&length=55)

Does this mean mem nodes will be forcefully grouped in an solo stream without even customer configuring it?

Reply:
A bit thinking into this is that we could extend configure file so that user could somehow specify the isolation of certain type(s) of nodes into a separate stream, hence in near future we do not have to constantly visit the code for changing requirements/cases.

Reply:
restored change